### PR TITLE
Bind the custom logger to the log methods, useful for loggers that use "this" on their own log methods

### PR DESCRIPTION
**Pull request by luislobo**
Bind the custom logger to the log methods, useful for loggers that use "this" on their own log methods

While using the latest winston 3.2.1, issues arise throwing error `self._addDefaultMeta is not a function`  as reported in https://github.com/balderdashy/sails/issues/4601

This fixes the issue

### CODE REVIEW

**Comment by luislobo** [comment]
Probably related to https://github.com/balderdashy/sails/issues/4337

**Comment by luislobo** [comment]
Created a project that uses winston 3.2.1, where the logger is working correctly with the current implementation

https://github.com/luislobo/sails-test-2019

Check https://github.com/luislobo/sails-test-2019/blob/master/config/log.js

File: index.js [hunk 1]
```diff
@@ -65,14 +65,14 @@ module.exports = function CaptainsLog(overrides) {
...
-    logger.warn = logger.warn || logger.error || nullLog;
-    logger.error = logger.error || nullLog;
-    logger.crit = logger.crit || logger.error || nullLog;
-    logger.verbose = logger.verbose || nullLog;
-    logger.silly = logger.silly || nullLog;
-    logger.blank = logger.blank || nullLog;
+    logger.debug = logger.debug ? logger.debug.bind(logger) : nullLog;
+    logger.info = logger.info ? logger.info.bind(logger) : nullLog;
+    logger.error = logger.error ? logger.error.bind(logger) : nullLog;
+    logger.warn = logger.warn ? logger.warn.bind(logger) : logger.error;
```
Comment (mikermcneil) [inline index.js]: looks like we kept the "error" fallback, but lost the nullLog short-circuit here

File: index.js [hunk 2]
```diff
@@ -65,14 +65,14 @@ module.exports = function CaptainsLog(overrides) {
...
+    logger.crit = logger.crit ? logger.crit.bind(logger) : logger.error;
```
Comment (mikermcneil) [inline index.js]: same here- kept the "error" fallback but lost the nullLog short-circuit

Comment (mikermcneil) [reply index.js]: also, I think we're losing the bind in our fallback -- maybe it should be `logger.crit? logger.crit.bind(logger) : logger.error? logger.error.bind(logger) : nullLog` ?

File: index.js (hunk 1 above)
Comment (mikermcneil) [reply index.js]: Also missing a `.bind` right?  so like `logger.warn? logger.warn.bind(logger) : logger.error? logger.error.bind(logger) : nullLog` ?

Comment (luislobo) [reply index.js]: In the `warn` method? well at the time `warn` is being set, `error` should be whether the `error` function if it exists, or the `nullLog` assigned in the `error` shortcut, am I correct?

Comment (luislobo) [reply index.js]: @mikermcneil In case you didn't see my comments up there

**Comment by ryd0rz** [comment]
Anything I can do to help move this PR along? Winston is preferred logging mechanism, and without this PR the latest version of winston not working.

I want to use winston so I can leverage sentry.io

**Comment by kierans** [comment]
@luislobo Are you able to tie this off? It would mean so much to a lot of people :)

File: index.js (hunk 2 above)
Comment (luislobo) [reply index.js]: @mikermcneil I see what you say, I've added the bindings

File: index.js (hunk 1 above)
Comment (luislobo) [reply index.js]: @mikermcneil I see what you say, I've added the bindings

**Comment by luislobo** [comment]
@mikermcneil This should be ready to review/merge.

All others, sorry about the delay, as always, open source development time is a luxury only a few can do full time ;)

**Comment by kierans** [comment]
@mikermcneil Are you able to look at this/merge it?

**Pull request merged**